data/post-router.js

```diff
@@ -0,0 +1,105 @@
+const express = require('express');
+
+const router = express.Router();
+
+const Posts = require('./db');
+
+router.get('/', async (req, res) => {
+    try {
+        const posts = await Posts.find(req.query);
+        res.status(200).json(posts);
+    } catch (error) {
+        console.log(error);
+        res.status(500).json({
+            message: 'The posts information could not be retrieved.',
+        });
+    }
+});
+
+router.get('/:id', async (req, res) => {
+    try {
+        const post = await Posts.findById(req.params.id);
+
+        if (post) {
+            res.status(200).json(post);
+        } else {
+            res.status(404).json({ message: 'Post not found' });
+        }
+    } catch (error) {
+        console.log(error);
+        res.status(500).json({
+            message: 'The posts information could not be retrieved.',
+        });
+    }
+});
+
+router.post('/', async (req, res) => {
+    try {
```

Comment:
you may want to check the data coming into the post here to insure its the correct type before calling the db